ea_restaurant_data_loader/lib/service/oauth2_service.ex

```diff
@@ -0,0 +1,80 @@
+defmodule EaRestaurantDataLoader.Oauth2Service do
+  alias EaRestaurantDataLoader.Repo
+  alias EaRestaurantDataLoader.AppClient
+  alias EaRestaurantDataLoader.AppClientScope
+  alias EaRestaurantDataLoader.AppAccessToken
+  alias EaRestaurantDataLoader.AppRefreshToken
+  alias EaRestaurantDataLoader.Oauth2Util
+  import Ecto.Query
+
+  def login_client(client_id, client_secret) do
+    secret_key = Application.get_env(:ea_restaurant_data_loader, :secret_key)
+
+    validate_client_credentials(client_id, client_secret, "ACTIVE")
+
+    client = get_client_by_client_id_and_entity_status(client_id, "ACTIVE")
+
+    scopes = get_scopes_by_client_id_and_entity_status(client.id, "ACTIVE")
+
+    access_token =
+      Oauth2Util.build_client_credentials_token(
+        client.client_name,
+        scopes.scope,
+        client.access_token_expiration_time,
+        secret_key
+      )
+
+    refresh_token =
+      Oauth2Util.build_client_credentials_token(
+        client.client_name,
+        scopes.scope,
+        client.refresh_token_expiration_time,
+        secret_key
+      )
+
+    {:ok, persisted_refresh_token} =
+      create_refresh_token(refresh_token, client, "CLIENT_CREDENTIALS")
+
+    {:ok, _} = create_access_token(access_token, persisted_refresh_token)
+
+    %{
+      :client_name => client.client_name,
+      :access_token => access_token,
+      :refresh_token => refresh_token,
+      :scopes => scopes.scope,
+      :expire_in => client.access_token_expiration_time
+    }
+  end
+
+  defp validate_client_credentials(client_id, client_secret, entity_status) do
+    client = get_client_by_client_id_and_entity_status(client_id, entity_status)
+
+    true = Bcrypt.verify_pass(client_secret, client.client_secret)
```

Comment:
Could you raise an error if this is false? A well defined error?

Reply:
Done.